bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -114,24 +114,45 @@ private void resetStatus() {
 		status = null;
 	}
 
+	@Override
+	public boolean isWholeWordSearchAvailable(String findString) {
+		if (!isWord(findString)) { // $NON-NLS-1$
+			return false;
+		}
+
+		return !isRegExSearchAvailableAndActive();
+	}
+
+	boolean isWholeWordSearchAvailableAndActive(String findString) {
+		if (isWholeWordSearchAvailable(findString)) {
+			return isActive(SearchOptions.WHOLE_WORD);
+		}
+		return false;
+	}
+
 	/**
-	 * Returns <code>true</code> if searching should be restricted to entire words,
-	 * <code>false</code> if not. This is the case if the respective checkbox is
-	 * turned on, regex is off, and the checkbox is enabled, i.e. the current find
-	 * string is an entire word.
+	 * Tests whether each character in the given string is a letter.
 	 *
-	 * @return <code>true</code> if the search is restricted to whole words
+	 * @param str the string to check
+	 * @return <code>true</code> if the given string is a word
+	 * @since 3.0
 	 */
-	private boolean isWholeWordSearchAvailableAndActive() {
-		return isActive(SearchOptions.WHOLE_WORD) && !isRegExSearchAvailableAndActive();
+	private boolean isWord(String str) {
+		if (str == null || str.isEmpty())
+			return false;
+
+		for (int i = 0; i < str.length(); i++) {
+			if (!Character.isJavaIdentifierPart(str.charAt(i)))
+				return false;
+		}
+		return true;
```

Comment:
This can be simplified:
```suggestion
return str != null && str.chars().allMatch(Character::isJavaIdentifierPart);
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/a0b224321663e63aa99f60da4a2527f993d68f7d